airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -789,6 +795,7 @@ def create_simple_retriever(
             stream_slicer=stream_slicer or SinglePartitionRouter(parameters={}),
             config=config,
             parameters=model.parameters,
+            disable_retries=self._disable_retries,
```

Comment:
I'm curious what the use case is for disabling retries when we are not using the `SimpleRetrieverTestReadDecorator`? Mainly because we used to be using that decorator for the test builder specific implementation details like max_slices, etc.

Reply:
This is a good question and I don't have a clear answer for this. I added this in the `SimpleRetriever` instead of the `SimpleRetrieverTestReadDecorator` because I could not find a clean way to do this using the current interface i.e. if we init `ModelToComponentFactory` with `disable_retries` but without `_limit_slices_fetched` or `_emit_connector_builder_messages`, how should it translate in terms of implementation?

Reply:
we talked about this a bit on Slack, given that we have to expose `disable_retries` config on the `ModelToComponentFactory`, it is likely more confusing to only honor this on the `SimpleRetrieverTestReadDecorator` So in that regard we can leave this as is on both the decoarator and `SimpleRetriever`.

However, this does feel like we're starting to be more apparent that all these settings start to make it less clear how they affect the components that are constructed as the list of settings, `limit_slices_fetched`, `page_limts`, `disable_retries` grows more. We're basically using the config of the factory as a surrogate for telling the factory this is a `TestRead` connector. But that goes a little outside the scope of this change and we don't need to block on this

Reply:
To be honest, I wouldn't mind having only one flag `test_read` and to configure everything based on that. It's not like we have different values for all of these and we only call it in one context. If we do that, we can simplify today and extract as needed later on